docs/csharp/language-reference/operators/and-operator.md

```diff
@@ -18,19 +18,21 @@ author: "BillWagner"
 ms.author: "wiwagn"
 ---
 # &amp; Operator (C# Reference)
-The & operator can function as either a unary or a binary operator.  
+The `&` operator can function as either a unary or a binary operator.  
   
 ## Remarks  
- The unary & operator returns the address of its operand (requires [unsafe](../../../csharp/language-reference/keywords/unsafe.md) context).  
+ The unary `&` operator returns the address of its operand (requires [unsafe](../../../csharp/language-reference/keywords/unsafe.md) context).  
   
- Binary & operators are predefined for the integral types and `bool`. For integral types, & computes the logical bitwise AND of its operands. For `bool` operands, & computes the logical AND of its operands; that is, the result is `true` if and only if both its operands are `true`.  
+ Binary `&` operators are predefined for the integral types and `bool`. For integral types, & computes the logical bitwise AND of its operands. For `bool` operands, & computes the logical AND of its operands; that is, the result is `true` if and only if both its operands are `true`.  
   
- The `&` operator evaluates both operators regardless of the first one's value. For example:  
+ The binary `&` operator evaluates both operators regardless of the first one's value, in contrast to the [conditional AND operator](../../../csharp/language-reference/operators/conditional-and-operator.md) `&&`. For example:  
   
  [!code-csharp[csRefOperators#37](../../../csharp/language-reference/operators/codesnippet/CSharp/and-operator_1.cs)]  
   
  User-defined types can overload the binary `&` operator (see [operator](../../../csharp/language-reference/keywords/operator.md)). Operations on integral types are generally allowed on enumeration. When a binary operator is overloaded, the corresponding assignment operator, if any, is also implicitly overloaded.  
   
+ The `&` operator can also be used as extremely efficient modulus for power-of-two divisors. `45 & 15` is equivalent to `45 % 16`, but as `%` is the [remainder operator](../../../csharp/language-reference/operators/remainder-operator.md), the results are different for negative dividends.  
```

Comment:
I don't think this paragraph is necessary here. Performance-focused folks know this fact, for others it might be more confusing. In general it's not description of *what* the operator does (the purpose of this article), but *what for* it might be used.

If you still want to keep this paragraph, please make it more clear:
- what is *extremely efficient*? Doesn't the compiler already optimize `a % b` to use bit mask when b is the power-of-two? What about [this comment](https://github.com/dotnet/csharplang/issues/1406#issuecomment-375085534)? If that's true, no need to use `&` for `%` at all.
- what definition of modulus do you use?
- looks like a typo: *`45 & 15` is equivalent to `45 % 16`*. Of course, it's not a typo but bit tricks. That also should be explained, as bit tricks might be not common knowledge, especially for the folks who learn about `&` operator.
- what is difference at negative cases?

Reply:
I originally had it more explained but I decided it was too long. The compiler does auto-optimize in this way when passed a `uint` but it is uncommon to see `uint` compared to `int`. The negative difference is explained on the `%` page, but perhaps this is not clear.

If you believe this part should be removed, I will.

Reply:
I think it can be removed.

Reply:
I agree with @pkulikov This paragraph can be removed. It's somewhat off topic.